src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +29,139 @@ ActorInit Obj_Wind_InitVars = {
+static ObjWindStruct D_80B2448C[] = {
+    { 530, 90, 3000, 2000, 0 },   { 1120, 100, 2000, 1000, 0 }, { 620, 100, 2000, 1000, 0 },
+    { 1620, 100, 2000, 1000, 0 }, { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 600, 120, 3000, 2000, 0 },  { 500, 80, 3000, 2000, 0 },
+    { 400, 60, 2000, 2000, 0 },   { 470, 70, 2000, 1000, 0 },
+};
+
+static s32 D_80B245CC = -1;
+
+void ObjWind_Update(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    ObjWindStruct* entry = &D_80B2448C[OBJ_WIND_GET_TYPE(thisx)];
+    Player* player;
+    f32 cosYTimesX;
+    f32 cosX;
+    f32 sinYTimesX;
+    Vec3f posCopy;
+    Vec3f nearestPoint;
+    Vec3f sp54;
+    f32 dist;
+    f32 distToNearestPoint;
+
+    if (D_80B245CC != DREG(85)) {
+        if ((DREG(85) >= 0) && (DREG(85) < ARRAY_COUNT(D_80B2448C))) {
+            DREG(80) = D_80B2448C[DREG(85)].unk_0;
+            DREG(81) = D_80B2448C[DREG(85)].unk_2;
+            DREG(82) = D_80B2448C[DREG(85)].unk_4;
+            DREG(83) = D_80B2448C[DREG(85)].unk_6;
+            DREG(84) = D_80B2448C[DREG(85)].unk_8;
+            D_80B245CC = DREG(85);
+        }
+    } else {
+        D_80B2448C[DREG(85)].unk_0 = DREG(80);
+        D_80B2448C[DREG(85)].unk_2 = DREG(81);
+        D_80B2448C[DREG(85)].unk_4 = DREG(82);
+        D_80B2448C[DREG(85)].unk_6 = DREG(83);
+        D_80B2448C[DREG(85)].unk_8 = DREG(84);
+    }
+    if (OBJ_WIND_GET_SWITCH_FLAG(thisx) == 0x7F || !Flags_GetSwitch(play, OBJ_WIND_GET_SWITCH_FLAG(thisx))) {
+        player = GET_PLAYER(play);
+        Math_Vec3f_Copy(&posCopy, &this->actor.world.pos);
+        cosX = Math_CosS(this->actor.shape.rot.x);
+        dist = Math_SinS(this->actor.shape.rot.x);
+        sinYTimesX = Math_SinS(this->actor.shape.rot.y) * dist;
+        cosYTimesX = Math_CosS(this->actor.shape.rot.y) * dist;
+        distToNearestPoint = func_80179A44(&posCopy, &player->actor.world, &nearestPoint);
+        if ((distToNearestPoint >= 0.0f) && (distToNearestPoint < entry->unk_0)) {
+            dist = Math_Vec3f_DistXYZAndStoreDiff(&player->actor.world.pos, &nearestPoint, &sp54);
+            if (dist < entry->unk_2) {
+                f32 var_fa0;
+                f32 windSpeedXZ = 1.0f; // reused temp
```

Comment:
Oh this is silly. So the `var_fa0 = windSpeedXZ - 1.0f;` is really just `0.0f`? Also you put the `// reused temp` comment here which is good, but could you also add it to `dist`?